incubator/command-dump/src/main/java/io/aklivity/zilla/runtime/command/dump/internal/airline/ZillaDumpCommand.java

```diff
@@ -318,253 +457,322 @@ private boolean handleFrame(
-
-                encodePcapHeader(buffer, ETHER_HEADER_SIZE + IPV6_HEADER_SIZE + TCP_HEADER_SIZE, timestamp);
-                encodeEtherHeader(buffer);
-                encodeIpv6Header(buffer, bindingId, streamId ^ 1L, streamId, TCP_HEADER_SIZE);
-                encodeTcpHeader(buffer, streamId ^ 1L, streamId, sequence, 0L, TcpFlag.FIN);
+                final MutableDirectBuffer buffer = new UnsafeBuffer(ByteBuffer.allocate(end.sizeof()));
+                final EndFW f = new EndFW.Builder().wrap(buffer, 0, end.sizeof()).set(end).build();
+                final ExtensionFW extension = f.extension().get(extensionRO::tryWrap);
+                patchExtension(buffer, extension, EndFW.FIELD_OFFSET_EXTENSION);
 
-                writePcapOutput(writer, buffer, 0, TCP_HEADER_LIMIT);
+                writeFrame(EndFW.TYPE_ID, f.originId(), f.routedId(), f.streamId(), f.timestamp(), f, PSH_ACK_FIN);
             }
         }
 
         private void onAbort(
             AbortFW abort)
         {
-            final long bindingId = abort.routedId();
+            if (allowedBinding.test(abort.routedId()))
+            {
+                final MutableDirectBuffer buffer = new UnsafeBuffer(ByteBuffer.allocate(abort.sizeof()));
+                final AbortFW f = new AbortFW.Builder().wrap(buffer, 0, abort.sizeof()).set(abort).build();
+                final ExtensionFW extension = f.extension().get(extensionRO::tryWrap);
+                patchExtension(buffer, extension, AbortFW.FIELD_OFFSET_EXTENSION);
 
-            if (allowedBinding.test(bindingId))
+                writeFrame(AbortFW.TYPE_ID, f.originId(), f.routedId(), f.streamId(), f.timestamp(), f, PSH_ACK_FIN);
+            }
+        }
+
+        private void onWindow(
+            WindowFW window)
+        {
+            if (allowedBinding.test(window.routedId()))
             {
-                final MutableDirectBuffer buffer = writeBuffer;
-                final long streamId = abort.streamId();
-                final long timestamp = abort.timestamp();
-                final long sequence = abort.sequence();
+                writeFrame(WindowFW.TYPE_ID, window.originId(), window.routedId(), window.streamId(), window.timestamp(), window,
+                    PSH_ACK);
+            }
+        }
 
-                encodePcapHeader(buffer, ETHER_HEADER_SIZE + IPV6_HEADER_SIZE + TCP_HEADER_SIZE, timestamp);
-                encodeEtherHeader(buffer);
-                encodeIpv6Header(buffer, bindingId, streamId ^ 1L, streamId, TCP_HEADER_SIZE);
-                encodeTcpHeader(buffer, streamId ^ 1L, streamId, sequence, 0L, TcpFlag.RST);
+        private void onReset(
+            ResetFW reset)
+        {
+            if (allowedBinding.test(reset.routedId()))
+            {
+                final MutableDirectBuffer buffer = new UnsafeBuffer(ByteBuffer.allocate(reset.sizeof()));
+                final ResetFW f = new ResetFW.Builder().wrap(buffer, 0, reset.sizeof()).set(reset).build();
```

Comment:
Please reuse member field flyweights instead of allocating per zilla frame.

Reply:
OK